src/es5.js

```diff
@@ -2,52 +2,78 @@
 // @ts-check
 
 import {
+    each,
+    has,
     is,
-    isProxyable,
-    PROXY_STATE,
+    isDraft,
+    isDraftable,
     shallowCopy,
-    RETURNED_AND_MODIFIED_ERROR,
-    has,
-    each,
-    finalize
+    DRAFT_STATE
 } from "./common"
 
 const descriptors = {}
-let states = null
 
-function createState(parent, proxy, base) {
-    return {
+// For nested produce calls:
+export const scopes = []
+export const currentScope = () => scopes[scopes.length - 1]
+
+export function willFinalize(result, baseDraft, needPatches) {
+    const scope = currentScope()
+    scope.forEach(state => (state.finalizing = true))
+    if (result === undefined || result === baseDraft) {
+        if (needPatches) markChangesRecursively(baseDraft)
+        // This is faster when we don't care about which attributes changed.
+        markChangesSweep(scope)
+    }
+}
+
+export function createDraft(base, parent) {
+    if (isDraft(base)) throw new Error("This should never happen. Please report: https://github.com/mweststrate/immer/issues/new") // prettier-ignore
+
+    const draft = shallowCopy(base)
+    each(base, prop => {
+        Object.defineProperty(draft, "" + prop, createPropertyProxy("" + prop))
+    })
+
+    const state = {
         modified: false,
```

Comment:
I think we get the build size down by enabling property mangling in the minifier, and just exclude the property names that are part of the public api. (Such as the public methods on an Immer instance).  Probably better to do in a separate PR though

Reply:
The draft state becomes a public API once hooks are added. ;)

Reply:
I just noticed. But that is only a subset of it correct? Anyway, an idea to experiment with, but not really related to this MR. Ill open fresh issue